### ISSUE TITLE
[Web] Cropped Picture to Image fails to load

### ISSUE THREAD

**Issue by deakjahn** [flutter/flutter#103803]
## Steps to Reproduce

**Web only!**

Use  the attached repro. It displays an image in the middle, cropped, using simple `dart.ui` operations. It used to work with CanvasKit although it has problems with DomCanvas (might be still related to this old https://github.com/flutter/engine/pull/20750). It stopped working with Flutter 3.0. Just removing the comment from the first line of the `croppedBitmap()` shows that it works with the full image.

[crop_repro_flutter.zip](https://github.com/flutter/flutter/files/8692942/crop_repro_flutter.zip)

**Expected results:** Image displayed in the center.

**Actual results:** Image not displayed.

<details>

```
E:\Android\flutter\bin\flutter.bat doctor --verbose
[√] Flutter (Channel stable, 3.0.0, on Microsoft Windows [Version 10.0.22000.675], locale en-US)
    • Flutter version 3.0.0 at E:\Android\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ee4e09cce0 (5 days ago), 2022-05-09 16:45:18 -0700
    • Engine revision d1b9a6938a
    • Dart version 2.17.0
    • DevTools version 2.12.2

[!] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at E:/Android/Sdk
    X cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    X Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/windows#android-setup for more details.

[X] Chrome - develop for the web (Cannot find Chrome executable at .\Google\Chrome\Application\chrome.exe)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.2.0)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.2.32505.173
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2020.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+9-b60-7590822)

[√] Android Studio (version 2021.2)
    • Android Studio at C:\Program Files\Android\Android Studio1
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)

[√] Connected device (2 available)
    • Windows (desktop) • windows          • windows-x64    • Microsoft Windows [Version 10.0.22000.675]
    • Edge (web)        • edge             • web-javascript • Microsoft Edge 101.0.1210.39

[√] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 2 categories.
Process finished with exit code 0
```

</details>

**Comment by maheshmnj** [comment]
Hi @deakjahn, Thanks for filing the issue. I am able to reproduce the issue on the stable and master channel. The image does not load on the Web (Without any error) But loads fine on the desktop.  Also when image is directly returned in the below function without crop

```dart
  Future<ui.Image> croppedBitmap(
      {required ui.Image bitmap,
      required Rect cropSize,
      ui.FilterQuality quality = FilterQuality.high}) async {
    //return bitmap; // When returned without crop image loads
    final recorder = ui.PictureRecorder();
    Canvas(recorder).drawImageRect(
      bitmap,
      cropSize,
      Offset.zero & cropSize.size,
      Paint()..filterQuality = quality,
    );
    return await recorder
        .endRecording()
        .toImage(cropSize.width.round(), cropSize.height.round());
  }
``` 


<img width="1358" alt="image" src="https://user-images.githubusercontent.com/31410839/168573210-92470006-30b1-428f-8ad0-4fb0d067338c.png">

Also,  I verified on `stable 2.2.3` and it doesn't reproduce there, so this is indeed a regression.

<img width="739" alt="image" src="https://user-images.githubusercontent.com/31410839/168574341-44e72fbf-eb3c-4c83-ac41-a7a4b5233087.png">


```
Flutter 2.2.3 • channel stable • https://github.com/flutter/flutter.git
Framework • revision f4abaa0735 (11 months ago) • 2021-07-01 12:46:11 -0700
Engine • revision 241c87ad80
Tools • Dart 2.13.4
```

<details>
<summary>flutter doctor -v (mac)</summary>

```
[✓] Flutter (Channel stable, 3.0.0, on macOS 12.3 21E230 darwin-arm, locale en-IN)
    • Flutter version 3.0.0 at /Users/mahesh/Documents/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ee4e09cce0 (6 days ago), 2022-05-09 16:45:18 -0700
    • Engine revision d1b9a6938a
    • Dart version 2.17.0
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-32, build-tools 31.0.0
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! CocoaPods 1.10.2 out of date (1.11.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your
        plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.66.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.40.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 12.3 21E230 darwin-arm
    • Chrome (web)    • chrome • web-javascript • Google Chrome 101.0.4951.64

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```

```
    • Flutter version 3.1.0-0.0.pre.419 at /Users/mahesh/Documents/flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 2eed8cbf93 (2 weeks ago), 2022-04-28 22:29:06 -0400
    • Engine revision dfdfe0b3b0
    • Dart version 2.18.0 (build 2.18.0-66.0.dev)
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-32, build-tools 31.0.0
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! CocoaPods 1.10.2 out of date (1.11.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your
        plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.66.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.40.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 12.3 21E230 darwin-arm
    • Chrome (web)    • chrome • web-javascript • Google Chrome 101.0.4951.64

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```
</details>

**Comment by kaumudpa** [comment]
Multiple cropping plugins are broken due to this, This need a quick fix!

**Comment by hterkelsen** [comment]
After some digging it seems the problem is `Picture.toImage()` not working correctly with drawing images, cropped or not. I've confirmed that `Picture.toImage()` works with pictures created from simple drawing commands like `drawRect`. Next I'll investigate if this is a problem in CanvasKit itself or the Flutter engine.

**Comment by jason-simmons** [comment]
This works as expected if `browserSupportsImageDecoder` is disabled.

(see https://github.com/flutter/flutter/issues/104030#issuecomment-1133518760)

**Comment by deakjahn** [comment]
Thanks.

**Comment by hterkelsen** [comment]
I was mistaken. The underlying problem is that uploading textures to CanvasKit and doing `toImage` on the picture is bugged on the Skia side. I am investigating the fix.

**Comment by itsjustkevin** [comment]
cc @itsjustkevin

**Comment by kaumudpa** [comment]
This is blocking release for multiple clients.

**Comment by rafaelmaia8384** [comment]
At the moment:

flutter run -d chrome --web-renderer canvaskit --dart-define=BROWSER_IMAGE_DECODING_ENABLED=false --profile

I hope this get solved quickly.

**Comment by kaumudpa** [comment]
> At the moment:
> 
> flutter run -d chrome --web-renderer canvaskit --dart-define=BROWSER_IMAGE_DECODING_ENABLED=false --profile
> 
> I hope this get solved quickly.

This flag does not work in debug mode.

**Comment by rafaelmaia8384** [comment]
> > At the moment:
> > flutter run -d chrome --web-renderer canvaskit --dart-define=BROWSER_IMAGE_DECODING_ENABLED=false --profile
> > I hope this get solved quickly.
> 
> This flag does not work in debug mode.

Use --profile or --release

**Comment by hterkelsen** [comment]
I have a fix for the Skia/CanvasKit side here: https://skia-review.googlesource.com/c/skia/+/545397

**Comment by kaumudpa** [comment]
Do we have a fix for this yet?

**Comment by orcecso** [comment]
hi, is there any updates regarding canvaskit?

**Comment by hterkelsen** [comment]
CanvasKit team is working on a fix: https://skia-review.googlesource.com/c/skia/+/557100

**Comment by CodeDoctorDE** [comment]
it's merged?

**Comment by orcecso** [comment]
it seems it is, but how do we gain access to it?


![image](https://user-images.githubusercontent.com/58863067/183790735-535c501e-664f-4a9c-ada8-4e50cce8bc03.png)

**Comment by kaumudpa** [comment]
This issue in not resolved in v3. 3.0

**Comment by njovy** [comment]
@kaumudpa Sadly this fix was merged two weeks ago so it wasn't included in 3.3.0.  I had to use the master branch to apply this fix and it works as expected.

**Comment by kaumudpa** [comment]
@njovy Its working perfectly on the master?

**Comment by njovy** [comment]
@kaumudpa Yes I needed ```PictureRecorder .toImage()```feature and it works like a charm on the master. Also the demo code displays an image on a browser.

**Comment by kaumudpa** [comment]
Not Merged on 3.3.1

**Comment by njovy** [comment]
@kaumudpa Flutter 3.3.x won't have this fix as the engine was fixed after 3.3

**Comment by deakjahn** [comment]
Nice. It broke apps in May and it's September now... :-)

**Comment by kaumudpa** [comment]
Also there should be a mechanism here to inform the community which issues are fixed in master already and which ones are yet land on the stable channel.

**Comment by pdeveltere** [comment]
Not fixed on stable as of 3.3.2 😢

**Comment by orcecso** [comment]
Upon checking, here are the results of me testing 3 versions on different channels, 

Master/Pre release
3.5.0-1.0.pre.7 - cropping is working

Beta
3.4.0-17.2.pre - cropping is working

Stable
3.3.0 - cropper results in white/gray picture only

may i ask on when or where can i find that flutter will push beta code to stable?

thanks
